### ISSUE TITLE
Another Crash Report (title subject to change)

### ISSUE THREAD

**Issue by Rewyn** [reteo/Survival-Industry#43]
http://openeye.openmods.info/crashes/6783a22bed39371e6d4a7efc577240a2

Shift+Rightclicked a shaft junction with a screwdriver, while reaching 80 or 90 fatigue at that moment, and getting the particles.

**Comment by jeffpengMC** [comment]
Having a look at the source I am now certain this has nothing to do with Somnia, but probably is entirely on Reika's end, most likely a syncing issue.

**Comment by reteo** [comment]
Do we have any mods that make use of movable frames?

**Comment by reteo** [comment]
Have you re-tested this issue, now that v10 is out?  Is it still happening?

**Comment by jeffpengMC** [comment]
TL;DR: At long last I can confirm this one is neither RoC nor Somnia. It's OptiFine's terribly sloppy multi-core chunk-loading. Solution: Use standard vanilla-style chunk-loading.

I've run into a similar issue just yesterday (this time with Special Mobs and Somnia) and it turned out to be the multi-core chunk loading OptiFine does when the corresponding option is set. A bit more analysis brought me to the following conclusion:

Chunkloading forces the client to render a specific area of the world, or better: forces the client to make a decision if a specified area is in the visible scope and has to be rendered or not and if so to render it. Minecrafts slower standard approach is to perform these loading actions stricly in succession to each other, while Optifine circumvents this and executes chunk loading parallel and thus out-of-order and without regard of inter-chunk dependency.

Now 2 things can happen with optifine: 1) The data depending on is not yet present. An explicit load cannot be triggered since the loading thread is no longer determinedly attached to the requesting thread. The result is a null pointer exception. 2) Two chunks try to "[tessalate](https://en.wikipedia.org/wiki/Tessellation)" at the same time, thus firing minecrafts "already tessalating" error, since their drawing results depend on each other.

Minecraft's standard logic would resolve such dependency problems by simply telling a chunk loading thread that it either has to wait on another thread, or that it has to make it's rendering decision without the other one. But since the entire execution is out of order, and optifine's algorythm doesn't really implement effective means to prevent such [race conditions](http://stackoverflow.com/questions/34510/what-is-a-race-condition), the whole thing ends in desaster.

Why does Somnia seem to be associated with this? The screen effect when suffering from sleep deprivation actually is triggered onto the screen changing the viewport thus forcing the client to re-evalute what it has to render. If now a chunk that did not previously require rendering enters the viewport and does depend on a chunk that already needs loading before processing.... the thing crashes.

**Comment by jeffpengMC** [comment]
As optifine is not part of the modpack and somewhat questionable to begin with..... we cannot and will not fix this.

**Comment by Rewyn** [comment]
Are you sure about that? I'm pretty damn sure I never added Optifine to any Survival Industry pack, I actually didn't use it for... quite a while now. Unless I'm missing the point?

**Comment by reteo** [comment]
Jeff, you might want to check the OpenEye report again; Optifine is not listed as one of the installed mods.

**Comment by reteo** [comment]
No further response has been added.  I'll assume this issue is resolved.  Reopen if this is not the case.
